pkg/reconciler/v1alpha1/taskrun/taskrun.go

```diff
@@ -155,35 +156,37 @@ func (c *Reconciler) Reconcile(ctx context.Context, key string) error {
 
 func (c *Reconciler) reconcile(ctx context.Context, tr *v1alpha1.TaskRun) error {
 	// get build the same as the taskrun, this is the value we use for 1:1 mapping and retrieval
-	b, err := c.getBuild(tr.Namespace, tr.Name)
+	//build, err := c.buildLister.Builds(tr.Namespace).Get(tr.Name)
+	build, err := c.buildLister.Builds(tr.Namespace).Get(tr.Name)
 	if errors.IsNotFound(err) {
-		if b, err = c.makeBuild(tr, c.Logger); err != nil {
-			return fmt.Errorf("Failed to create a build for taskrun: %v", err)
+		// Build is not present, create build
+		build, err = c.createBuild(tr)
+		if err != nil {
+			c.Logger.Errorf("Failed to create build for task %q :%v", err, tr.Name)
+			return err
 		}
 	} else if err != nil {
-		return fmt.Errorf("Failed retrieving build %s for taskRun %s: %v", tr.Name, tr.Name, err)
+		c.Logger.Errorf("Failed to reconcile taskrun: %q, failed to get build %q; %v", tr.Name, tr.Name, err)
+		return err
+	}
+
+	// TODO(aaron-prindle) verify that build does not have to exist to create a tracker
```

Comment:
not sure I understand this comment. We verify the build exists above, or we create it right?

Reply:
That comment existed before I started the PR. Looks like its deleted upstream